src/extensions/default/Inline3DParametersEditor/InlineParameterEditor.js

```diff
@@ -0,0 +1,205 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var InlineWidget         = brackets.getModule("editor/InlineWidget").InlineWidget,
+        ParameterEditor      = require("ParameterEditor").ParameterEditor;
+
+    /** @type {number} Global var used to provide a unique ID for each parameter editor instance's _origin field. */
+    var lastOriginId = 1;
+
+    /**
+     * Inline widget containing a ParameterEditor control
+     * @param {!string} parameters  Initial Parameters
+     * @param {!CodeMirror.TextMarker} marker
+     */
+    function InlineParameterEditor(parameters, marker) {
+        this._parameters = parameters;
+        this._marker = marker;
+        this._isOwnChange = false;
+        this._isHostChange = false;
+        this._origin = "+InlineParameterditor_" + (lastOriginId++);
+
+        this._handleParametersChange = this._handleParametersChange.bind(this);
+        this._handleHostDocumentChange = this._handleHostDocumentChange.bind(this);
+        InlineWidget.call(this);
+    }
+
+    InlineParameterEditor.prototype = Object.create(InlineWidget.prototype);
+    InlineParameterEditor.prototype.constructor = InlineParameterEditor;
+    InlineParameterEditor.prototype.parentClass = InlineWidget.prototype;
+
+    InlineParameterEditor.prototype.ParameterEditor = null;
+
+    InlineParameterEditor.prototype._parameters = null;
+
+    /**
+     * Range of code we're attached to; _marker.find() may by null if sync is lost.
+     * @type {!CodeMirror.TextMarker}
+     */
+    InlineParameterEditor.prototype._marker = null;
+
+    /** @type {boolean} True while we're syncing a parameter picker change into the code editor */
+    InlineParameterEditor.prototype._isOwnChange = null;
+
+    /** @type {boolean} True while we're syncing a code editor change into the parameter picker */
+    InlineParameterEditor.prototype._isHostChange = null;
+
+    /** @type {number} ID used to identify edits coming from this inline widget for undo batching */
+    InlineParameterEditor.prototype._origin = null;
+
+
+    /**
+     * Returns the current text range of the parameters we're attached to, or null if
+     * we've lost sync with what's in the code.
+     * @return {?{start:{line:number, ch:number}, end:{line:number, ch:number}}}
+     */
+    InlineParameterEditor.prototype.getCurrentRange = function () {
+        var pos, start, end;
+
+        pos = this._marker && this._marker.find();
+
+        start = pos && pos.from;
+        if (!start) {
+            return null;
+        }
+
+        end = pos.to;
+        if (!end) {
+            end = {line: start.line};
+        }
+
+        var line = this.hostEditor.document.getLine(start.line),
+            matches = line.substr(start.ch).match(/((-)?\d+(\.\d+)?) ((-)?\d+(\.\d+)?) ((-)?\d+(\.\d+)?)/g);
```

Comment:
Let's add a comment above this that specifies what you're looking for here using an example.